### ISSUE TITLE
Leading or Trailing whitespace in profile 'id' element causes NPE in maven-git-versioning-extension

### ISSUE THREAD

**Issue by JWT007** [qoomon/maven-git-versioning-extension#239]
Hello Bengt,

unfortunately I have a new problem for you.

It has a simple workaround so low prio :)

If the profile id in the POM contains a leading or trailing whitespace, the maven build blows up spectacularly.

```
<project ...>
  <profiles>
    <profile>
      <id>foobar </id>
    </profile>
  </profiles>
</project>
```

```
[17:02:04.428+0100] [ERROR] Internal error: java.lang.NullPointerException: Cannot invoke "org.apache.maven.model.ModelBase.getProperties()" because "model" is null -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.NullPointerException: Cannot invoke "org.apache.maven.model.ModelBase.getProperties()" because "model" is null
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:108)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:821)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:270)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.maven.model.ModelBase.getProperties()" because "model" is null
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.lambda$updatePropertyValues$78 (GitVersioningModelProcessor.java:1185)
    at java.util.ArrayList.forEach (ArrayList.java:1511)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.updatePropertyValues (GitVersioningModelProcessor.java:1184)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.updateProfiles (GitVersioningModelProcessor.java:1313)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.writePomFile (GitVersioningModelProcessor.java:1158)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.processModel (GitVersioningModelProcessor.java:301)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.read (GitVersioningModelProcessor.java:120)
    at org.apache.maven.model.building.DefaultModelBuilder.readModel (DefaultModelBuilder.java:552)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:257)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:243)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:446)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:409)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:366)
    at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:349)
    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:340)
    at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:76)
    at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:453)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:196)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:172)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:100)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:821)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:270)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47) 
```

Removing the whitespace fixes the problem.

**Comment by JWT007** [comment]
NOTE: I don't urgently need a fix for this so for me totally ok to bundle into future release with other fixes :P

**Comment by qoomon** [comment]
fixed see version `9.6.3`

**Comment by qoomon** [comment]
@JWT007 and thanks a lot for reporting
